etl/embedding_model_pipelines/embedding_model_pipeline.py

```diff
@@ -0,0 +1,12 @@
+from abc import ABC, abstractmethod
+
+from langchain.docstore.document import Document
+from langchain_core.vectorstores import VectorStore
+
+
+class EmbeddingModelPipeline(ABC):
+    """An interface to build embedding model pipelines that transform Records into embeddings."""
+
+    @abstractmethod
+    def create_embedding_store(self, documents: tuple[Document, ...]) -> VectorStore:
+        """Create an embedding store."""
```

Comment:
If you're going to have a docstring it should be more descriptive than the method name.

Reply:
I have improved the docstring here and elsewhere.